.travis.yml

```diff
@@ -3,24 +3,28 @@ language: rust
 env:
   - RUSTFLAGS="-D warnings"
 
-before_script:
-  - rustup component add rustfmt
-
 matrix:
   fast_finish: true
   include:
     - rust: nightly
       os: linux
-      env: BUILD_DOCS=1
+      env: BUILD_DOCS=1 BUILD_BOOK=1
     - rust: nightly
       os: osx
       osx_image: xcode9.2
       env: BUILD_DOCS=1
     - rust: nightly-x86_64-pc-windows-msvc
       os: windows
 
+before_script:
+  - rustup component add rustfmt
+  - (test -x $HOME/.cargo/bin/cargo-install-update || cargo install cargo-update)
+  - (test -x $HOME/.cargo/bin/mdbook || cargo install --vers "^0.3" mdbook)
+  - cargo install-update -a
+
 script:
-  - cargo check --all --benches --bins --examples --tests
-  - cargo test --all
+  - if ![[ -n "$BUILD_BOOK" ]]; then cargo check --all --benches --bins --examples --tests && cargo test --all; fi
+  - if  [[ -n "$BUILD_BOOK" ]]; then cargo test  --all --benches --bins --examples --tests; fi
```

Comment:
I've had to modify the travis config to where, when the book should be built, we don' run `cargo check` but instead just run `cargo test`.

This might cause ci to not fail as quickly on the linux build but the problem is that running `cargo check` followed by `cargo test` produces two entries for `async_std` in `./target/debug/deps` which then causes `mdbook test` to fail when trying to resolve `extern crate async_std`.

I don't know of another way to fix that issue. If this is an acceptable workaround then I think this PR is good to go.